tests/integration/new_relations/helpers.py

```diff
@@ -42,8 +42,34 @@ async def build_connection_string(
     )
     host = endpoints.split(",")[0].split(":")[0]
 
+    # Translate the service hostname to an IP address.
+    cmd = [
+        "sg",
+        "microk8s",
+        "-c",
+        " ".join(
+            [
+                "microk8s.kubectl",
+                "get",
+                "service",
+                "-n",
+                ops_test.model_name,
+                host.split(".")[0],
+                "-o",
+                "json",
+                "|",
+                "jq",
+                "-r",
+                ".spec.clusterIP",
```

Comment:
Consider using lightkube here to fetch the service so that you don't accidentally tie our tests to microk8s?

Reply:
Thanks Jon. Good suggestion. I will update it.

Reply:
Implemented on https://github.com/canonical/postgresql-k8s-operator/pull/12/commits/70d1347458d5e7e2792f0c4447d331ff6bdb2047.